frontend/src/components/contactsForm.tsx

```diff
@@ -0,0 +1,310 @@
+  const [email, setEmail] = useState<string>('')
+  const [phoneNumber, setPhoneNumber] = useState<string>('')
+  const [notes, setNotes] = useState<string>('')
+  const [errorMessage, setErrorMessage] = useState<string>('')
+  const [showError, setShowError] = useState<boolean>(false)
+  const [showForm, setShowForm] = useState<boolean>(true)
+
+  const resetForm = () => {
+    setFirstName('')
+    setLastName('')
+    setEmail('')
+    setPhoneNumber('')
+    setNotes('')
+    setErrorMessage('')
+    setRelationshipType('')
+    setEventDate('')
+    setEventType('')
+    setRelationships([])
+    setImportantDates([])
+  }
+
+  const handleClose = () => {
+    setShowForm(false)
+    resetForm()
+    setShowCreateAContactForm(false)
+  }
+
+  const handleChange =
+    (setState: React.Dispatch<React.SetStateAction<string>>) =>
+      (event: React.ChangeEvent<HTMLInputElement>) => {
+        setState(event.target.value)
+      }
+  const handleFirstNameChange = handleChange(setFirstName)
+  const handleLastNameChange = handleChange(setLastName)
+  const handleEmailChange = handleChange(setEmail)
+  const handlePhoneNumberChange = handleChange(setPhoneNumber)
+  const handleNotesChange = handleChange(setNotes)
+
+  const addRelationship = () => {
+    if (relationshipType.trim() === '') {
+      setShowError(true)
+      return
+    }
+    setRelationships((prevRelationships) => [
+      ...prevRelationships,
+      {
+        relationshipType
+      }
+    ])
+    setRelationshipType('')
+  }
+
+  const handleRelationshipChange = (
+    index: number,
+    field: keyof Relationship,
+    value: any
+  ) => {
+    setRelationships((prevRelationships) =>
+      prevRelationships.map((type, i) =>
+        i === index ? { ...type, [field]: value } : type
+      )
+    )
+  }
+
+  const handleRemoveRelationshipType = (index: number) => {
+    setRelationships((prevRelationships) => [
+      ...prevRelationships.slice(0, index),
+      ...prevRelationships.slice(index + 1)
+    ])
+  }
+
+  const addImportantEvent = () => {
+    if (eventDate.trim() === '' || eventType.trim() === '') {
+      setShowError(true)
+      return
+    }
+    setImportantDates((prevDates) => [
+      ...prevDates,
+      {
+        date: eventDate,
+        eventType
+      }
+    ])
+    setEventDate('')
+    setEventType('')
+  }
+
+  const handleImportantEventChange = (
+    index: number,
+    field: keyof ImportantDate,
+    value: any
+  ) => {
+    setImportantDates((prevDates) =>
+      prevDates.map((date, i) =>
+        i === index ? { ...date, [field]: value } : date
+      )
+    )
+  }
+
+  const handleRemoveImportantEvent = (index: number) => {
+    setImportantDates((prevDates) => [
+      ...prevDates.slice(0, index),
+      ...prevDates.slice(index + 1)
+    ])
+  }
+
+  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault()
+    try {
+      await axios.post('http://localhost:8000/contacts', {
```

Comment:
reminder that this url will not work once we deploy the app. add a todo here to change this url